client/landing/stepper/declarative-flow/internals/steps-repository/importer/index.tsx

```diff
@@ -97,18 +104,26 @@ export function withImporterWrapper( Importer: ImporterCompType ) {
 			dispatch( requestSites() );
 		}, [ dispatch ] );
 
-		useEffect( () => {
-			! siteId && site?.ID && setSiteId( site.ID );
-		}, [ siteId, site ] );
 		useAtomicTransferQueryParamUpdate( siteId );
 		useEffect( fetchImporters, [ siteId ] );
 		useEffect( checkFromSiteData, [ fromSiteData?.url ] );
 		useEffect( () => onComponentUnmount, [] );
 
-		if ( ! importer ) {
-			stepNavigator.goToImportCapturePage?.();
-			return null;
-		}
+		useEffect( () => {
+			if ( ! isLoading && ! site ) {
+				logToLogstash( {
+					feature: 'calypso_client',
+					tags: [ 'importer', importer, 'error' ],
+					error: 'Importer missing site info',
+					message: 'Importer missing site',
+					site_id: siteId,
+					site_slug: siteSlug,
+					properties: {
+						env: config( 'env_id' ),
+					},
+				} );
+			}
+		}, [ importer, isLoading, site, siteId ] );
```

Comment:
Nit: we're using `siteSlug` in the code, so we should list it as a dep.
```suggestion
		}, [ importer, isLoading, site, siteId, siteSlug ] );
```

Reply:
Fixed 8cdbf88f74b6ccd256549534a676ac2c12a4988f